apps/comps/hooks/useProfile.ts

```diff
@@ -41,12 +42,17 @@ export const useProfile = () => {
  */
 export const useUpdateProfile = () => {
   const queryClient = useQueryClient();
+  const { trackEvent } = useAnalytics();
 
   return useMutation({
     mutationFn: async (data: UpdateProfileRequest) => {
       return apiClient.updateProfile(data);
     },
-    onSuccess: () => {
+    onSuccess: (_, variables) => {
+      trackEvent("UserUpdatedProfile", {
+        updated_fields: Object.keys(variables),
```

Comment:
technically, this will include whatever fields are defined in the original type, even if they're undefined.

we could do something like:
```ts
const updatedFields = Object.keys(variables).filter(
  (key) => variables[key as keyof UpdateProfileRequest] !== undefined,
);
```